server/utils/rrAuthenticate.js

```diff
@@ -1,11 +1,23 @@
 import * as rrAuth from '../api/rrAuth';
-const { USERNAME, PASSWORD, CLIENT_ID, CLIENT_SECRET, TOKEN_FILE } = process.env;
+import { ENVIRONMENT_VARS, DEFAULT_ENVIRONMENT } from '../constants/rrconstants';
 import path from 'path';
 import fs from 'fs';
 import jwt from 'jsonwebtoken';
+import { printFuncError, printFuncWarning, printFuncLog } from '../utils/utils';
 
-export function rrAuthenticate() {
+// In order to support multiple orgs, the rrAuthenticate function can be called with an orgID
+// which will load a org that may differ from the default.
+export function rrAuthenticateWithSubdomain(subdomain) {
+    return rrAuthenticate(ENVIRONMENT_VARS.find(env => env.SUBDOMAIN == subdomain).ID);
+}
+
+// Generic auth function, will load org credentials from rrconstants file, if orgId is empty
+// it will use the 'Default Environment ID'
+export function rrAuthenticate(orgId = DEFAULT_ENVIRONMENT) {
     return new Promise((resolve, reject) => {
+        const ENV = ENVIRONMENT_VARS.find(env => env.ID == orgId);
```

Comment:
just in case if ENV is not found you could default to empty object or just do an IF statement that it is not falsy right after
```suggestion
        const ENV = ENVIRONMENT_VARS.find(env => env.ID == orgId) || {};
```

Reply:
updated.